adapters/repos/db/vector/composer/config.go

```diff
@@ -0,0 +1,107 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package composer
+
+import (
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/weaviate/weaviate/adapters/repos/db/vector/common"
+	"github.com/weaviate/weaviate/adapters/repos/db/vector/flat"
+	"github.com/weaviate/weaviate/adapters/repos/db/vector/hnsw"
+	"github.com/weaviate/weaviate/adapters/repos/db/vector/hnsw/distancer"
+	"github.com/weaviate/weaviate/entities/cyclemanager"
+	"github.com/weaviate/weaviate/entities/errorcompounder"
+	"github.com/weaviate/weaviate/entities/schema"
+	ent "github.com/weaviate/weaviate/entities/vectorindex/composer"
+	"github.com/weaviate/weaviate/usecases/monitoring"
+)
+
+type Config struct {
+	ID                       string
+	TargetVector             string
+	Logger                   logrus.FieldLogger
+	RootPath                 string
+	ShardName                string
+	ClassName                string
+	PrometheusMetrics        *monitoring.PrometheusMetrics
+	VectorForIDThunk         common.VectorForID[float32]
+	TempVectorForIDThunk     common.TempVectorForID
+	DistanceProvider         distancer.Provider
+	MakeCommitLoggerThunk    hnsw.MakeCommitLogger
+	TombstoneCallbacks       cyclemanager.CycleCallbackGroup
+	ShardCompactionCallbacks cyclemanager.CycleCallbackGroup
+	ShardFlushCallbacks      cyclemanager.CycleCallbackGroup
+}
+
+func (c Config) Validate() error {
```

Comment:
Should we also validate async indexing is enabled if it is a requirement for the composer index?